server/planning/planning/planning_spike_async.py

```diff
@@ -0,0 +1,192 @@
+
+    # TODO-ASNYC: Convert this to async call when function is added to PlanningAsyncService
+    get_resource_service("planning").delete_assignments_for_coverages(
+        assignments_to_delete, notify_user_on_failed_assignment_deletes
+    )
+
+
+async def notify_draft_coverage_on_spike(coverage: dict[str, Any]):
+    assignment_service = AssignmentResourceModel.get_service()
+
+    assigned_to = coverage.get("assigned_to")
+    if assigned_to and assigned_to.get("assignment_id"):
+        user = get_user()
+        assignment = await assignment_service.find_by_id_raw(assigned_to["assignment_id"])
+        if assignment:
+            slugline = assignment.get("planning", {}).get("slugline", "")
+            coverage_type = assignment.get("planning", {}).get("g2_content_type", "")
+            PlanningNotifications().notify_assignment(
+                coverage_status=coverage.get("workflow_status"),
+                target_user=assignment.get("assigned_to", {}).get("user"),
+                target_desk=(
+                    assignment.get("assigned_to", {}).get("desk")
+                    if not assignment.get("assigned_to", {}).get("user")
+                    else None
+                ),
+                message="assignment_spiked_msg",
+                slugline=slugline,
+                coverage_type=get_coverage_type_name(coverage_type),
+                actioning_user=user.get("display_name", user.get("username", "Unknown")),
+                omit_user=True,
+            )
+
+
+async def process_spike_planning_item(updates: dict[str, Any], original: dict[str, Any]) -> dict[str, Any]:
+    """
+    Function to spike planning item.
+
+    :param updates: The update payload from the client.
+    :param original: The original planning document.
+    :return: The updated planning document.
+    """
+    planning_service = PlanningResourceModel.get_service()
+
+    if original.get("pubstatus") or original.get("state") not in [
+        WORKFLOW_STATE.INGESTED,
+        WORKFLOW_STATE.DRAFT,
+        WORKFLOW_STATE.POSTPONED,
+        WORKFLOW_STATE.CANCELLED,
+    ]:
+        raise SuperdeskApiError.badRequestError(message="Spike failed. Planning item in invalid state for spiking.")
+
+    user = get_user()
+
+    updates["revert_state"] = original[ITEM_STATE]
+    updates[ITEM_STATE] = WORKFLOW_STATE.SPIKED
+    set_item_expiry(updates)
+
+    coverages = deepcopy(original.get("coverages") or [])
+    for coverage in coverages:
+        if coverage.get("workflow_status") == WORKFLOW_STATE.ACTIVE:
+            coverage["workflow_status"] = WORKFLOW_STATE.DRAFT
+            coverage["assigned_to"] = {}
+
+    updates["coverages"] = coverages
+
+    # Mark item as unlocked directly in order to avoid more queries and notifications
+    # coming from lockservice.
+    remove_lock_information(updates)
+
+    remove_autosave_on_spike(original)
```

Comment:
I looked up [this function](https://github.com/superdesk/superdesk-planning/blob/async/server/planning/common.py#L391), and it's used only by Event and Planning spike functions.
We should now be able to update this function so it is async and uses the async autosave resources

Reply:
This is true.

However, this function is also used by `PlanningSpikeService` which is used by `EventsPostService` which is yet to be converted to async. 

Hence why I removed the `PlanningUnspikeService` only cause it wasn't used anywhere else. But retained the planning spike as an internal resource for this. 

I've added more context to the todo. I think it will be safe to convert it when `EventPostService` is changed to async.

Reply:
Fair enough :+1: